packages/core/src/StateNode.ts

```diff
@@ -1307,11 +1307,16 @@ class StateNode<
         action
       ): action is
         | RaiseActionObject<TEvent>
-        | SendActionObject<TContext, TEvent, TEvent> =>
-        action.type === actionTypes.raise ||
-        (action.type === actionTypes.send &&
+        | SendActionObject<TContext, TEvent, TEvent> => {
+        const isSendAction = action.type === actionTypes.send;
+        const isRaisableSendAction =
+          isSendAction &&
           (action as SendActionObject<TContext, TEvent>).to ===
-            SpecialTargets.Internal)
+            SpecialTargets.Internal &&
+          !(action as SendActionObject<TContext, TEvent>).delay;
+
+        return action.type === actionTypes.raise || isRaisableSendAction;
+      }
```

Comment:
nit: if this is a `raise` action then it's not worth checking upfront if it's a raisable send action
```ts
        if (action.type === actionTypes.raise) {
            return true;
        }
        if (action.type !== actionTypes.send) {
            return false;
        }
        const sendAction = action as SendActionObject<TContext, TEvent>;
        return sendAction.to === SpecialTargets.Internal || !sendAction.delay;
```